### ISSUE TITLE
Proposal: Relax requirement on "raw" values in pipeline for multiplication and division

### ISSUE THREAD

**Issue by crertel** [rrrene/credo#434]
### Environment

* Credo version (`mix credo -v`): **0.8.6**
* Erlang/Elixir version (`elixir -v`): **1.3.4**

### What were you trying to do?

At various times, credo gets upset if you do something like:

```
1.0 * foo |> Float.round(3)
```

And would prefer something like:

```
1.0 |> Kernel.*(foo) |> Float.round(3)
```

### Expected outcome

Credo gets upset.

### Actual outcome

Credo shouldn't get upset, since piping a number into `Kernel.*` just looks silly when compared with basic arithmetic.

**Comment by devonestes** [comment]
Would writing that as `Float.round(1.0 * foo, 3)` be an ok solution for this example? Personally I think that reads really well, and it would make Credo totally happy.

**Comment by crertel** [comment]
My example truncates a longer pipeline used in production code.

**Comment by devonestes** [comment]
Hmmm, well in that context it does make a little more sense. Otherwise, the only way to get a raw value to start that pipeline would be to do:

```elixir
total_price = quantity_of_thing * price_of_thing
total_price
|> Float.round(3)
|> do_other_thing_with_subtotal
|> adjust_price_for_tidal_forces( Externalities.get_moon() )
|> additional_patchup_munging
```

In my eyes that's not all that bad, but I can see why others might not like it.

**Comment by rrrene** [comment]
If anyone wants to craft a PR for this, I am all for it! :+1:

**Comment by crertel** [comment]
I'll take a look at it when I get a chance. Haven't worked on static analysis tooling yet, so this could be fun. :)
